base/number.jl

```diff
@@ -136,6 +136,11 @@ true
 """
 signbit(x::Real) = x < 0
 
+ispositive(x) = !isnonnegative(x)
+isnegative(x) = signbit(x) && !iszero(x)
```

Comment:
Why not 
```suggestion
isnegative(x) = !isnonpositive(x)
```
as you did above?

Reply:
because there is one more operation? tbh I don't know if Julia does the optimization.

Reply:
> tbh I don't know if Julia does the optimization.

I'm away from computer, but I'm pretty sure it'd optimise away double negations. And can you please also comment on the second on the second question. I don't really see the need for 4 new functions when 2 would be enough.

Reply:
> it'd optimise away double negations

If that's the case then 2 is enough.

Reply:
Wait, "nonpositive" does not mean "negative"; it means "negative or zero" (and similarly for "nonnegative"). So, there still might be some value in providing the definitions
```julia
isnonpositive(x) = isnegative(x) || iszero(x)
isnonnegative(x) = ispositive(x) || iszero(x)
```